packages/lfo-input/src/LFOInput.ts

```diff
@@ -0,0 +1,188 @@
+  handleEachInput,
+  HedronEngine,
+  InputOptionNodesConfig,
+  IPlugin,
+  NodeValue,
+} from '@hedron/engine'
+
+const TAU = Math.PI * 2
+const lerp = (v0: number, v1: number, t: number) => (1 - t) * v0 + t * v1
+
+export class LFOInput implements IPlugin {
+  public readonly id = 'lfo-input'
+  public readonly name = 'LFO Input'
+  public readonly inputType = 'lfo'
+  public readonly description =
+    'Generates LFO waves (e.g. sin, square, sawtooth) as inputs for params.'
+  public readonly optionNodesConfig = [
+    {
+      key: 'isEnabled',
+      valueType: 'boolean',
+      defaultValue: true,
+    },
+    {
+      key: 'frequency',
+      valueType: 'enum',
+      defaultValue: 1,
+      options: [
+        {
+          value: 32,
+          label: '32',
+        },
+        {
+          value: 16,
+          label: '16',
+        },
+        {
+          value: 8,
+          label: '8',
+        },
+        {
+          value: 4,
+          label: '4',
+        },
+        {
+          value: 2,
+          label: '2',
+        },
+        {
+          value: 1,
+          label: '1',
+        },
+        {
+          value: 1 / 2,
+          label: '1/2',
+        },
+        {
+          value: 1 / 4,
+          label: '1/4',
+        },
+        {
+          value: 1 / 8,
+          label: '1/8',
+        },
+        {
+          value: 1 / 16,
+          label: '1/16',
+        },
+        {
+          value: 1 / 32,
+          label: '1/32',
+        },
+      ],
+    },
+    {
+      key: 'waveType',
+      valueType: 'enum',
+      defaultValue: 'sine',
+      options: [
+        { value: 'sine', label: 'Sine' },
+        { value: 'square', label: 'Square' },
+        { value: 'sawtooth', label: 'Sawtooth' },
+        { value: 'triangle', label: 'Triangle' },
+      ],
+    },
+    {
+      key: 'amplitude',
+      valueType: 'number',
+      defaultValue: 1,
+    },
+    {
+      key: 'phase',
+      valueType: 'number',
+      defaultValue: 0,
+    },
+    {
+      key: 'min',
+      valueType: 'number',
+      defaultValue: 0,
+    },
+    {
+      key: 'max',
+      valueType: 'number',
+      defaultValue: 1,
+    },
+  ] as const satisfies InputOptionNodesConfig
+
+  private inputLatches: Record<string, boolean> = {}
+
+  constructor(engine: HedronEngine) {
+    const clock = engine.clock
+
+    if (!clock) {
+      throw new Error('Clock plugin is required for LFOInput to function.')
+    }
+
+    const store = engine.getStore()
+
+    const tick = () => {
+      requestAnimationFrame(() => {
+        const storeState = store.getState()
+        handleEachInput<typeof this.optionNodesConfig>(
+          storeState,
+          'lfo',
+          ({ input, optionNodes: opts, targetNode }) => {
+            // TODO: This can be handled by `onInput` once we have `isEnabled` as a generic option
+            if (!opts.isEnabled) return
+
+            const delta = clock.beatDelta * opts.frequency * TAU + opts.phase
```

Comment:
```suggestion
            const delta = (clock.beatDelta * opts.frequency + opts.phase) * TAU
```
have phase complete a full loop

Reply:
whoops, nice catch! Thanks!